clang/test/SemaSYCL/device_global.cpp

```diff
@@ -33,6 +33,65 @@ namespace {
 device_global<int> same_name; // OK
 }
 
+// TODO: add diagnostic for this case
+inline namespace other {
+device_global<int> same_name; // ILLEGAL: shadows "device_global" variable
+} // namespace other
+
+// TODO: add diagnostic for this case
+// inline namespace {
+//  namespace foo {               // ILLEGAL: namespace name shadows "::foo"
+//  }                             // namespace which contains "device_global"
+//                                // variable.
+//}
+
+struct BBar {
+private:
+  struct BarInsider {
+    static device_global<float> c;
+  };
+};
+
+struct ABar {
+  void method() {
+    // expected-error@+1{{'device_global' variables must be static or declared at namespace scope}}
+    static device_global<float> c;
+  }
+  struct BarInsider {
+    static device_global<float> c;
+    void method() {
+      // expected-error@+1{{'device_global' variables must be static or declared at namespace scope}}
```

Comment:
isn't this testing the same thing as test in ABar?

Reply:
Technically, it is showing a device_global in a method in a nested struct, which ABar isn't.